### ISSUE TITLE
Terrain importer broken in version 1.6

### ISSUE THREAD

**Issue by McSpider** [Zylann/godot_heightmap_plugin#324]
**Describe the bug**
Terrain importer not working.

**To Reproduce**
Attempting to import terrain using the `Terrain>Import maps...` menu raises several `Invalid call. Nonexistent function` errors.

**Cause**
Errors are somewhat nonsensical. Fix for me was to remove `: HT_Logger` static typing from the following lines:
https://github.com/Zylann/godot_heightmap_plugin/blob/9ec3e913280f7d6c6bf95f55c0b606071c799d32/addons/zylann.hterrain/tools/importer/importer_dialog.gd#L212
https://github.com/Zylann/godot_heightmap_plugin/blob/9ec3e913280f7d6c6bf95f55c0b606071c799d32/addons/zylann.hterrain/tools/importer/importer_dialog.gd#L231

I'm not sure why this is an issue and if removing it is the correct course of action, but doing so allows me to use the importer.

**Environment**
Windows 10
Godot 3.3.3 to 3.4.4
Plugin version 1.6

**Other Notes**
Not tested in Godot 3.3.2 or lower due to the add-on using `.gdshader` files, which weren't introduced until 3.3.3.

**Comment by Zylann** [comment]
> Fix for me was to remove : HT_Logger static typing from the following lines:

This is indeed complete nonsense... I can't fix that this way. Godot is buggy here.

**Comment by McSpider** [comment]
So I just did a clean download of Godot 3.4.4, new project, installed add-on through AssetLib, restarted Godot, activated the addon and setup a test scene. Same problem. Restarted Godot, still broken. Then I repeated the steps above but got the addon directory by cloning it from GitHub. Same errors:
```
 res://addons/zylann.hterrain/tools/importer/importer_dialog.gd:260 - Invalid call. Nonexistent function 'debug' in base 'Nil'.
 res://addons/zylann.hterrain/tools/importer/importer_dialog.gd:189 - Invalid call. Nonexistent function 'has' in base 'Nil'.
 res://addons/zylann.hterrain/tools/importer/importer_dialog.gd:87 - Invalid get index 'errors' (on base: 'Nil').
```
Also tested this in two different machines a Mac and a Windows PC, both behaved the same.

I did check around some more, the only cases where you're static typing the `HT_logger` is in `importer_dialog.gd`

And then I noticed that in `logger.gd` two classes are defined; `HT_LoggerBase `& `HT_LoggerVerbose` so I tried changing the `: HT_Logger` to `: HT_Logger.HT_LoggerBase` and that makes everything work fine. I believe this is because in Godot, scenes and scripts are inherently classes. So doing `HT_Logger.get_for(self)` actually returns `HT_LoggerBase` or `HT_LoggerVerbose` which can be accessed using `HT_Logger.HT_LoggerBase` in the importer script. 

I feel like this somewhat makes sense aside from the fact that it works for you without modification.

**Comment by Zylann** [comment]
> And then I noticed that in logger.gd two classes are defined; HT_LoggerBase & HT_LoggerVerbose

Oh, that would be that then. So indeed the static typing in this specific case must be changed or removed because it's not the right class.
Yet... I think Godot is still buggy for not raising that all along when I tested this xD
I believe I didnt get the error because """""""""static"""""""" typing did absolutely nothing when the wrong logger type was passed to the function, instead it triggers when a function is called on it. Which... did not happen for me, because I tried importing a PNG without errors. You must have been importing something else like a RAW (which does print debug stuff), or had errors/warnings.

**Comment by McSpider** [comment]
Yeah Godot 3 could definitely do better with the error messages. I was indeed importing RAW files, didn't even occur to me to try a PNG... Would you like me to open up a pull request for changing `HT_Logger` to `HT_Logger.HT_LoggerBase`?

**Comment by Zylann** [comment]
I'm currently doing the changes, preparing for a 1.6.1 release

**Comment by Zylann** [comment]
It should be fixed in 6f269ad2190f9a5cb02656830eb200a38176191f, I submitted an edit to the AssetLibrary so 1.6.1 should be available there later.

**Comment by McSpider** [comment]
Cool, thanks for the help in getting this figured out.
